taichi/runtime/llvm/llvm_runtime_executor.cpp

```diff
@@ -394,12 +394,13 @@ void LlvmRuntimeExecutor::initialize_llvm_runtime_snodes(
   const int tree_id = field_cache_data.tree_id;
   const int root_id = field_cache_data.root_id;
 
+  root_size =
```

Comment:
Why is this necessary? IIUC `iroundup` below will do this if root_size != 0?

Reply:
In which circumstance will root_size be 0?

Reply:
hmmm, good point. `iroundup` actually rounds it to zero with `root_size == 0`, let me see if it's an implementation bug.

Reply:
`iroundup` should round it to zero with `root_size=0`. Do you need to allocate any memory if `root_size=0`?